src/raftstore/store/snap/io.rs

```diff
@@ -68,17 +68,22 @@ pub fn build_sst_cf_file(
 ) -> Result<BuildStatistics, Error> {
     let mut sst_writer = create_sst_file_writer(snap, cf, path)?;
     let mut stats = BuildStatistics::default();
-    let mut base = if io_limiter.is_some() { 0 } else { usize::MAX };
+    let base = io_limiter
+        .as_ref()
+        .map_or(0 as i64, |l| l.get_max_bytes_per_time());
+    let mut bytes: i64 = 0;
     box_try!(snap.scan_cf(cf, start_key, end_key, false, |key, value| {
+        let entry_len = key.len() + value.len();
         if let Some(ref io_limiter) = io_limiter {
-            if stats.total_size >= base {
-                let next_request = io_limiter.get_max_bytes_per_time();
-                io_limiter.request(next_request);
-                base += next_request as usize;
+            // seems the usage of io_limiter here do not match its design
```

Comment:
How should we use it?

Reply:
I think it should be something like `while(bytes>=base){bytes-=base;...}`